README.md

```diff
@@ -1,2 +1,203 @@
+
+- ROOT_MODULES
+
+The directory with all the module manifest (.yml) files
+
+Currently the installation procedure makes use of [rootbase](https://github.com/root-project/root) source files for root packages and modules.
+- rootbase package install procedure
+
+The routine uses package specific manifest file and installs each module from the manifest.
+- rootbase module install procedure
+
+The routine requires the module specific manifest file and proceeds with it.
+
+root-get extensively makes use of cache to avoid re-building of entities. The already built packages are stored in cache directory.
+
+## External entity handling routine
+
+- External entity - Entity outside rootbase scope.
+
+The entity can be present in user system or may not be present on the system. For external entity also, root-get requires manifest file.
+The routine checks for entity presence in user system otherwise uses downloader routine to download the entity repository from GitHub(currently supported) and returns the path for the new downloaded directory. The further process takes the similar path as per internal entity routine.
+
+User can provide own manifest file with all neccessary attributes present in it to plug an external entity. To create your own manifest file please follow the instructions below
+
+- Creating module manifest file :
+```
+module:
+  name: 'modulename'
+  packageurl: 'link/to/github/repository'
+  tag: 0.0.0 
+  path: 'path/to/module/in/system'
+  publicheaders: 'if present'
+  sources: 'if present'
+  targets: 'modulename'
+  deps: 'dependencies'
+```
+
+- Example module.yml :
+```
+module: 
+  name: FFTW
+  packageurl: "https://github.com/root-project/root/tree/master/math/fftw"
+  tag: 0.0.0 
+  path: /home/username/root/math/fftw
+  publicheaders: 
+  sources: no_sources
+  targets: FFTW
+  deps: Core
+```  
+
+- Creating package manifest file :
+```
+package:
+  name: 
+  tag: 'version value'
+  targets:
+    target:
+    name: 'packagename'
+  products:
+    package:
+      name: 
+  module1:
+    name: 'modulename'
+    packageurl: 'link/to/github/repository'
+    tag: 0.0.0 
+    path: 'path/to/module/in/system'
+    publicheaders: 'if present'
+    sources: 'if present'
+    targets: 'modulename'
+    deps: 'dependencies'
+  module2:
+    .
+    .
+  module:
+```
+
+- Example packagename.yml :
+```
+package: 
+ name: "IO"
+ tag: 0.0.0
+ targets:
+  target:
+   name: "IO"
+ products:
+  package:
+   name: IO
+ module: 
+  name: "RCastor"
+  packageurl: "https://github.com/root-project/io/castor"
+  tag: 0.0.0
+  path: /home/ashwin/root/io/castor
+  publicheaders: 
+  sources: no_sources
+  targets: 
+  deps: Net;Core;RIO
+ module: 
+  name: "DCache"
+  packageurl: "https://github.com/root-project/io/dcache"
+  tag: 0.0.0
+  path: /home/ashwin/root/io/dcache
+  publicheaders: 
+  sources: no_sources
+  targets: 
+  deps: Net;Core;RIO 
+```
+
+## Prerequisites
+
+python3
+
+[matplotlib](https://matplotlib.org/users/installing.html)
```

Comment:
what is a command that user will need to use to install matplotlib and networkx?

Reply:
The links are directed to official installation instructions only.